addons/html_editor/static/src/core/delete_plugin.js

```diff
@@ -470,10 +471,15 @@ export class DeletePlugin extends Plugin {
 
     fillShrunkBlocks(commonAncestor) {
         const fillBlock = (block) => {
-            if (block === this.editable) {
+            if (
+                block.getAttribute("contenteditable") === "true" &&
+                !block.parentElement.isContentEditable &&
+                !paragraphRelatedElements.includes(block.tagName)
```

Comment:
Not sure about `paragraphRelatedElements`, maybe we should use a whitelist instead? We wouldn't want to put a `p` inside a block which can only have phrasing content
I think we can start with only `div` for now and increase the whitelist as needed

Reply:
Changed it. But did not start an actual list with one element.
(we can always start one if we come to need it for something else than a div)